packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  $,
+  Slot,
+  component$,
+  createContextId,
+  useContext,
+  useContextProvider,
+  useSignal,
+  useStylesScoped$,
+  type ContextId,
+  type QRL,
+  type Signal,
+} from '@builder.io/qwik';
+import { type JSXNode } from '@builder.io/qwik/jsx-runtime';
+import CSS from './portal-provider.css?inline';
+
+// Define public API for opening up Portals
+export const PortalAPI = createContextId<
+  /**
+   * Add JSX to a portal.
+   * @param name portal name.
+   * @param jsx to add.
+   * @param contexts to add to the portal.
+   * @returns A function used for closing the portal.
+   */
+  QRL<
+    (name: string, jsx: JSXNode, contexts?: ContextPair<any>[]) => () => void
+  >
+>('PortalManagerAPI');
+
+export type ContextPair<T> = { id: ContextId<T>; value: T };
+
+// Define public API for closing Portals
+export const PortalCloseAPI =
+  createContextId<QRL<() => void>>('PortalCloseAPI');
+
+// internal context for managing portals
+const Portals = createContextId<Signal<Portal[]>>('Portals');
```

Comment:
Same comment here with the suffix of ContextId

Reply:
👍